packages/main/src/AINotice.ts

```diff
@@ -0,0 +1,175 @@
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import ResizeHandler from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import type { ResizeObserverCallback } from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import i18n from "@ui5/webcomponents-base/dist/decorators/i18n.js";
+import AINoticeTemplate from "./generated/templates/AINoticeTemplate.lit.js";
+import AINoticeCss from "./generated/themes/AINotice.css.js";
+import { AINOTICE_BUTTON_TOOLTIP } from "./generated/i18n/i18n-defaults.js";
+
+import Link from "./Link.js";
+import Button from "./Button.js";
+import Popover from "./Popover.js";
+import Icon from "./Icon.js";
+
+/**
+ * @class
+ *
+ * ### Overview
+ * The `AINotice` component is a custom UI5 web component that displays an informational notice.
+ * It can include text for attribution and verification, as well as an optional icon. The layout
+ * of the component adapts based on its content and available space.
+ *
+ * ### Usage
+ * Use the `AINotice` component when you need to display an informational message with the option to include attribution, verification, and an icon.
+ *
+ * **Properties:**
+ * - `attributionText`: The text for attribution (e.g., source or author).
+ * - `verificationText`: The text for verification (e.g., "Verified", "Approved").
+ * - `showIcon`: A boolean indicating whether the icon should be shown.
+ *
+ * **Note**: The layout dynamically adjusts based on the available space and the content's width. If space is limited, verification text may be hidden to maintain the layout.
+ *
+ * ### ES6 Module Import
+ * `import "@ui5/webcomponents/dist/AINotice.js";`
+ *
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @since 2.4.0
+ */
+
+@customElement({
+	tag: "ui5-ainotice",
+	renderer: litRender,
+	template: AINoticeTemplate,
+	styles: AINoticeCss,
+	dependencies: [Link, Button, Popover, Icon],
+})
+
+class AINotice extends UI5Element {
+	/**
+	 * A callback function used for resizing logic.
+	 * @private
+	 */
+	_handleResize: ResizeObserverCallback;
+
+	/**
+	 * Defines the attribution text to be displayed in the component.
+	 * @public
+	 */
+	@property()
+	attributionText = "";
+
+	/**
+	 * Defines the verification text to be displayed in the component.
+	 * @public
+	 */
+	@property()
+	verificationText = "";
+
+	/**
+	 * Defines whether an icon should be shown in the component.
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	showIcon = false;
+
+	/**
+	 * The popup slot allows for custom popups to be inserted into the component.
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	popup!: HTMLElement;
```

Comment:
The slots are array by nature, users can slot multiple elements inside, even if the component works and expects a single popup. Thus, the proper TS type would be

popup!: Array<HTMLElement>;